cmake/kernel.cmake

```diff
@@ -0,0 +1,238 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Zephyr Kernel CMake module.
```

Comment:
Alternative naming ideas to 'kernel', about which I agree is not the best name:

- core (already mentioned in another comment)
- base